ngraph_bridge/ngraph_builder.cc

```diff
@@ -2564,26 +2565,129 @@ static Status TranslatePadOp(const Node* op,
+static Status TranslatePadV2Op(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input, ng_paddings_op, ng_constant_values;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_paddings_op,
+                                   &ng_constant_values));
+
+  std::vector<int64> paddings;
+  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &paddings));
+
+  NGRAPH_VLOG(3) << "{" << ng::join(paddings) << "}";
+
+  if (paddings.size() % 2 != 0) {
+    return errors::InvalidArgument(
+        "Constant node for paddings does not have an even number of "
+        "elements");
+  }
+
+  ng::CoordinateDiff padding_before(paddings.size() / 2);
+  ng::CoordinateDiff padding_after(paddings.size() / 2);
+  auto pad_mode = ng::op::PadMode::CONSTANT;
+
+  for (size_t i = 0; i < paddings.size() / 2; i++) {
+    padding_before[i] = paddings[2 * i];
+    padding_after[i] = paddings[2 * i + 1];
+  }
+
+  NGRAPH_VLOG(3) << "{" << ng::join(padding_before) << "}";
+  NGRAPH_VLOG(3) << "{" << ng::join(padding_after) << "}";
+
+  auto pads_begin_node = make_shared<ng::opset3::Constant>(
+      ng::element::i64, ng::Shape{padding_before.size()}, padding_before);
+  auto pads_end_node = make_shared<ng::opset3::Constant>(
+      ng::element::i64, ng::Shape{padding_after.size()}, padding_after);
+
+  auto pad_op = ConstructNgNode<ng::opset3::Pad>(op->name(), ng_input,
+                                                 pads_begin_node, pads_end_node,
+                                                 ng_constant_values, pad_mode);
+
+  SaveNgOp(ng_op_map, op->name(), pad_op);
+  return Status::OK();
+}
+
+// See https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/mirror-pad
+static Status TranslateMirrorPadOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input, ng_paddings_op;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_paddings_op));
+
+  std::vector<int64> paddings;
+  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &paddings));
+  NGRAPH_VLOG(3) << "{" << ng::join(paddings) << "}";
+
+  std::string pad_mode_str;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "mode", &pad_mode_str));
+  ng::op::PadMode pad_mode = ng::op::PadMode::CONSTANT;  // Impossible Value
+  if (pad_mode_str == "REFLECT") {
+    pad_mode = ng::op::PadMode::REFLECT;
+  } else if (pad_mode_str == "SYMMETRIC") {
+    pad_mode = ng::op::PadMode::SYMMETRIC;
+  } else {
+    return errors::InvalidArgument(pad_mode_str,
+                                   " is not an allowed padding mode.");
+  }
+
+  if (paddings.size() % 2 != 0) {
+    return errors::InvalidArgument(
+        "Constant node for paddings does not have an even number of "
+        "elements");
+  }
+
+  ng::CoordinateDiff padding_before(paddings.size() / 2);
```

Comment:
You can put this common part into a helper function to construct `opset3::Pad` that all 3 pad ops can use.